runtime/compiler/control/JITaaSCompilationThread.cpp

```diff
@@ -2603,19 +2587,27 @@ remoteCompile(
       resolvedMirrorMethodsPersistIPInfo = std::get<7>(recv);
       if (statusCode >= compilationMaxError)
          throw JITaaS::StreamTypeMismatch("Did not receive a valid TR_CompilationErrorCode as the final message on the stream.");
+      if (statusCode != compilationStreamVersionIncompatible)
```

Comment:
We could store this in the stream itself. As long as the stream is not recreated we have the version check already done.